### ISSUE TITLE
windows UNC paths not working

### ISSUE THREAD

**Issue by Jacq** [dokufreaks/plugin-filelist#1]
Windows UNC paths: \\examplepath\mypath are not supported by this plugin.
Also in order to support resource access in firefox,iexplorer,chrome... ftp links should be provided instead of direct links to the file.
Finally showsize and showdate is not available for list.
I add here a unidiff that fix UNC paths and provide these features. A new parameter is added to the plugin list 'ftp'. If ftp=1 the ftp link is provided for the resource. For example for a \\server\mypath\myresource a ftp://server/mypath/myresource link is provided:

```
--- syntaxold.php	lun nov 16 19:23:00 2009

+++ /syntaxnew.php	mié feb 16 00:39:57 2011

@@ -91,6 +91,7 @@
             'direct' => 0,
             'recursive' => 0,
             'titlefile' => '_title.txt',
+			'ftp' => 0,
         );
         foreach($flags as $flag) {
             list($name, $value) = split('=', $flag);
@@ -124,10 +125,10 @@
 
             // if we got nothing back, display a message
             if ($result == DOKU_PLUGIN_FILELIST_NOMATCH) {
-                $renderer->doc .= '[n/a: ' . $this->getLang('error_nomatch') . ']';
+                $renderer->doc .= '[n/a: ' . utf8_encode($this->getLang('error_nomatch')) . ']';
                 return true;
             } else if ($result == DOKU_PLUGIN_FILELIST_OUTSIDEJAIL) {
-                $renderer->doc .= '[n/a: ' . $this->getLang('error_outsidejail') . ']';
+                $renderer->doc .= '[n/a: ' . utf8_encode($this->getLang('error_outsidejail')) . ']';
                 return true;
             }
 
@@ -200,11 +201,16 @@
         $link['more']   = '';
         $link['class']  = 'media';
         if (!$params['direct']) {
-            $link['url'] = ml(':'.str_replace('/', ':', substr($filepath, strlen($this->mediadir))));
+			$link['url'] = ml(':'.str_replace('/', ':', substr($filepath, strlen($this->mediadir))));			
         } else {
             $link['url'] = $webdir.substr($filepath, strlen($basedir));
+			if ($params['ftp'])
+			{
+			$link['url'] = 'ftp:' . utf8_encode(str_replace('\\','/',$link['url']));			
+			}
         }
-        $link['name']   = $filename;
+
+        $link['name']   = utf8_encode($filename);
         $link['title']  = $renderer->_xmlEntities($link['url']);
         if($conf['relnofollow']) $link['more'] .= ' rel="nofollow"';
 
@@ -239,6 +245,7 @@
      * @return void
      */
     function _render_list_items($files, $basedir, $webdir, $params, &$renderer, $level = 1) {
+	global $conf;
         if ($params['style'] == 'olist') {
             $renderer->listo_open();
         } else {
@@ -249,13 +256,21 @@
             if ($file['children'] !== false && $file['treesize'] > 0) {
                 // render the directory and its subtree
                 $renderer->listitem_open($level);
-                $renderer->doc .= $file['name'];
+                $renderer->doc .= utf8_encode($file['name']);			    
                 $this->_render_list_items($file['children'], $basedir, $webdir, $params, $renderer, $level+1);
                 $renderer->listitem_close();
             } else if ($file['children'] === false) {
                 // render the file
                 $renderer->listitem_open($level);
                 $this->_render_link($file['name'], $file['path'], $basedir, $webdir, $params, $renderer);
+				if ($params['tableshowdate']) 
+				{					
+					$renderer->doc .= ', (' . strftime($conf['dformat'], $file['mtime']). ')'; 					
+				} 
+				if ($params['tableshowsize']) 
+				{
+					$renderer->doc .= ', ' . $this->_size_readable($file['size'], 'PiB', 'bi', '%01.1f %s');
+				}
                 $renderer->listitem_close();
             } else {
                 // ignore empty directories
@@ -285,18 +300,18 @@
 
         if ($params['tableheader']) {
             $renderer->tableheader_open();
-            $renderer->doc .= $this->getLang('filename');
+            $renderer->doc .= utf8_encode($this->getLang('filename'));
             $renderer->tableheader_close();
 
             if ($params['tableshowsize']) {
                 $renderer->tableheader_open();
-                $renderer->doc .= $this->getLang('filesize');
+                $renderer->doc .= utf8_encode($this->getLang('filesize'));
                 $renderer->tableheader_close();
             }
 
             if ($params['tableshowdate']) {
                 $renderer->tableheader_open();
-                $renderer->doc .= $this->getLang('lastmodified');
+                $renderer->doc .= utf8_encode($this->getLang('lastmodified'));
                 $renderer->tableheader_close();
             }
 
@@ -411,6 +426,7 @@
                 $pattern = DOKU_INC.$pattern;
             }
         }
+
         // get the canonicalized basedir (without resolving symlinks)
         $dir = $this->_realpath($this->_win_path_convert(dirname($pattern))).'/';
 
@@ -693,9 +709,10 @@
     function _realpath($path) {
         $path=explode('/', $path);
         $output=array();
-        for ($i=0; $i<sizeof($path); $i++) {
-            if (('' == $path[$i] && $i > 0) || '.' == $path[$i]) continue;
-            if ('..' == $path[$i] && $i > 0 && '..' != $output[sizeof($output) - 1]) {
+        for ($i=0; $i<sizeof($path); $i++) {		
+        //  if (('' == $path[$i] && $i > 0) || '.' == $path[$i]) continue;
+			if ('.' == $path[$i]) continue;
+            if ('..' == $path[$i] && '..' != $output[sizeof($output) - 1]) {
                 array_pop($output);
                 continue;
             }
@@ -763,6 +780,7 @@
     /**
      * Determines whether a given path is absolute or relative.
      * On windows plattforms, it does so by checking whether the second character
+	 * ADDED: Also if the path is a network shared that starts with //
      * of the path is a :, on all other plattforms it checks for a / as the
      * first character.
      *
@@ -771,7 +789,7 @@
      */
     function _path_is_absolute($path) {
         if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
-            return ($path[1] == ':');
+            return ($path[1] == ':') || (($path[0] == '/') && ($path[1] == '/'));
         } else {
             return ($path[0] == '/');
         }
```

**Comment by LarsGit223** [comment]
> Finally showsize and showdate is not available for list.

This is now implemented, see #16.

**Comment by LarsGit223** [comment]
An option to generate FTP links has been implemented, see #18.

**Comment by LarsGit223** [comment]
All your changes have been implemented now - so I close this.
Thanks a lot for your contribution!
